crates/node/src/shims/abcipp_shim.rs

```diff
@@ -177,6 +191,62 @@ impl AbcippShim {
             }
         }
     }
+
+    fn update_snapshot_task(&mut self, take_snapshot: TakeSnapshot) {
+        if self
+            .snapshot_task
+            .as_ref()
+            .map(|t| t.is_finished())
+            .unwrap_or_default()
+        {
+            let task = self.snapshot_task.take().unwrap();
+            match task.join() {
+                Ok(Err(e)) => tracing::error!(
+                    "Failed to create snapshot with error: {:?}",
+                    e
+                ),
+                Err(e) => tracing::error!(
+                    "Failed to join thread creating snapshot: {:?}",
+                    e
+                ),
+                _ => {}
+            }
+        }
+        let TakeSnapshot::Yes(db_path) = take_snapshot else {
+            return;
+        };
+        let base_dir = self.service.base_dir.clone();
+
+        let (snap_send, snap_recv) = tokio::sync::oneshot::channel();
+        let snapshot_task = std::thread::spawn(move || {
+            let db = crate::storage::open(db_path, true, None)
+                .expect("Could not open DB");
+            let snapshot = db.snapshot();
+            // signal to main thread that the snapshot has finished
+            snap_send.send(()).unwrap();
```

Comment:
can we comment something here to explain why this is required? I believe it is because we can't take a snapshot of some inconsistent state, only of fully committed state.

ideally we wouldn't need to block reads though. we only need to block db writes while snapshots are taking place. we could modify the rocksdb impl to block writes while a snapshot is being taken, but allow reads to proceed as usual. then we could delete this channel, and instead set some flag e.g. a `block_db_writes: AtomicBool` in rocksdb or smth

Reply:
I agree that ideally we wouldn't block state reads. I'm not sure if it's possible, but rocksdb claims making a snapshot is fast.

Reply:
let's leave this pending for now, then